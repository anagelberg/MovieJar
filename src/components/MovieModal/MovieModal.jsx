import "./MovieModal.scss";
import Modal from "../Modal/Modal";
import MovieSummary from "../MovieSummary/MovieSummary";
import { ReactComponent as CloseIcon } from '../../assets/icons/close.svg'
import EditDelIcons from "../EditDelIcons/EditDelIcons";

function MovieModal({ show, movie, delClick, closeHandler }) {

    return (
        <Modal
            show={show}
            headerContent={() => {
                return (
                    <>
                        <div className='movie-modal__edit-del'>
                            <EditDelIcons delClick={delClick} />
                        </div>
                        <CloseIcon className="movie-modal__close" onClick={closeHandler} aria-label="Close" />
                    </>
                )
            }}
            bodyContent={() => <MovieSummary movie={movie} />}
        />
    );
}

export default MovieModal;