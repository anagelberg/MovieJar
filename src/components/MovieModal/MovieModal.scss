@use "../../styles/_mixins.scss" as *;
@use "../../styles/_variables.scss" as *;

.movie-modal {
  &__head {
    width: 100%;

    @include flex-center($justify-content: space-between);
  }

  &__close {
    fill: $dark-text;
    @include small-icon;
  }
}
