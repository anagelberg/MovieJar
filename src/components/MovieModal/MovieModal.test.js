import { render, fireEvent, screen } from '@testing-library/react';
import MovieModal from './MovieModal'; // Adjust the import path as necessary
import '@testing-library/jest-dom';

describe('MovieModal Component', () => {
    const mockCloseHandler = jest.fn();
    const mockDelClick = jest.fn();
    const mockMovie = {
        title: "The Godfather"
    }

    it('renders correctly when show is true', () => {
        render(<MovieModal show={true} movie={mockMovie} delClick={mockDelClick} closeHandler={mockCloseHandler} />);
        expect(screen.queryByRole('dialog')).toBeInTheDocument();
        expect(screen.queryByText(mockMovie.title)).toBeInTheDocument();
    });

    it('does not render when show is false', () => {
        render(<MovieModal show={false} movie={mockMovie} delClick={mockDelClick} closeHandler={mockCloseHandler} />);
        expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
        expect(screen.queryByText(mockMovie.title)).not.toBeInTheDocument();

    });

    it('calls closeHandler when close icon is clicked', () => {
        render(<MovieModal show={true} movie={mockMovie} delClick={() => { }} closeHandler={mockCloseHandler} />);
        fireEvent.click(screen.getByLabelText('Close'));
        expect(mockCloseHandler).toHaveBeenCalled();
    });

    it('matches snapshot', () => {
        const { container } = render(<MovieModal show={true} movie={mockMovie} delClick={mockDelClick} closeHandler={mockCloseHandler} />);
        expect(container).toMatchSnapshot();
    });
});

