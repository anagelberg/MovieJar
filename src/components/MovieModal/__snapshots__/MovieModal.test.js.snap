// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieModal Component matches snapshot 1`] = `
<div>
  <div
    class="modal"
    role="dialog"
  >
    <div
      class="modal__content"
    >
      <div
        class="modal__top"
      >
        <div
          class="modal__header"
        >
          <div
            class="movie-modal__edit-del"
          >
            <div
              class="icons"
            >
              <svg
                class="icons__del"
              >
                delete.svg
              </svg>
              <svg
                class="icons__edit"
              >
                edit.svg
              </svg>
            </div>
          </div>
          <svg
            aria-label="Close"
            class="movie-modal__close"
          >
            close.svg
          </svg>
        </div>
        <div
          class="modal__body"
        >
          <div
            class="summary__text"
          >
            <div>
              <div
                class="summary__title-container"
              >
                <h4
                  class="summary__title"
                >
                  The Godfather
                </h4>
                <h4
                  class="summary__title"
                >
                  (
                  )
                </h4>
              </div>
              <p
                class="rating"
              >
                undefined / 5
                <img
                  alt="star"
                  class="rating__icon"
                  src="star.svg"
                />
              </p>
            </div>
            <p
              class="summary__description"
            />
            <p
              class="summary__runtime"
            >
               minutes
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
